kubernetes-client/src/main/java/io/fabric8/kubernetes/client/informers/SharedInformerFactory.java

```diff
@@ -347,7 +346,12 @@ private static String getKeyStrForField(String str) {
     return keyBuilder.toString();
   }
 
-  private <T extends HasMetadata, L extends KubernetesResourceList<T>> BaseOperation<T, L, ?> getConfiguredBaseOperation(String namespace, OperationContext context, Class<T> apiTypeClass, Class<L> apiListTypeClass) {
+  private static <T> boolean isKeyOfType(String key, Class<T> apiTypeClass) {
+    String plural = CustomResource.getPlural(apiTypeClass);
```

Comment:
Calling `CustomResource#getPlural` from here is a clear symptom that this method should be moved to `HasMetadata`.

Proper steps would be to move the method to HasMetadata class, keep the method in CustomResource class and call the newer in HasMetadata while marking it as deprecated with instructions to call the new one.

Reply:
Agreed, initially I thought that it only made sense on `CustomResource` but it makes also sense on `HasMetadata` when I think more about it.

Reply:
Done.